# This file was put together/created/updated by Augusto Bott <augusto at bott com br> 
# as a contribution to the Alt-F project. Alt-F provides an alternative firmware 
# for the D-LINK DNS-323 NAS. You are free to use, copy, distribute and modify it 
# under the terms of the GNU GPL, see /COPYING for details.
#
# this file should be put on ${SRCDIR}/toolchain/kernel-headers
#
# see http://www.kernel.org/pub/linux/kernel/v2.6/ChangeLog-2.6.36
# commit 445f12dca8f06113caa775315bff574dee72b27d
#
--- a/arch/arm/mach-orion5x/include/mach/system.h
+++ b/arch/arm/mach-orion5x/include/mach/system.h
@@ -25,6 +25,8 @@ static inline void arch_reset(char mode, const char *cmd)
	 */
	orion5x_setbits(RSTOUTn_MASK, (1 << 2));
	orion5x_setbits(CPU_SOFT_RESET, 1);
+	mdelay(200);
+	orion5x_clrbits(CPU_SOFT_RESET, 1);
 }
